packages/core-components/src/testUtils.ts

```diff
@@ -25,19 +32,118 @@
  *  https://mui.com/material-ui/react-use-media-query/#testing
  *
  * @public
+ *
+ * @xample
+ * Match with any media query:
+ * ```ts
+ * mockBreakpoint({ matches: true });
+ * ```
+ *
+ * @xample
+ * Set the initial breakpoint:
+ * ```ts
+ * mockBreakpoint({ initialBreakpoint: 'md' });
+ * ```
+ *
+ * @example
+ * Map media queries to breakpoints:
+ * ```ts
+ * mockBreakpoint({ queryBreakpointMap: { '(min-width:1500px)': 'xl', '(min-width:1000px)': 'lg', '(min-width:700px)': 'md', '(min-width:400px)': 'sm', '(min-width:0px)': 'xs', } });
+ * ```
+ *
+ * @example
+ * Change the breakpoint during the test:
+ * ```ts
+ * const { set } = mockBreakpoint({ initialBreakpoint: 'md' });
+ * set('lg');
+ * ```
  */
-export function mockBreakpoint(options: { matches: boolean }) {
+export function mockBreakpoint(
+  options:
+    | {
+        matches: boolean;
+      }
+    | {
+        initialBreakpoint?: Breakpoint;
+        /** Defaults to { '(min-width:1920px)': 'xl', '(min-width:1280px)': 'lg', '(min-width:960px)': 'md', '(min-width:600px)': 'sm', '(min-width:0px)': 'xs' } */
+        queryBreakpointMap?: Record<string, Breakpoint>;
+      },
+): {
+  set(value: string): void;
+  remove(): void;
+} {
+  const mediaQueries: {
+    mediaQueryString: string;
+    mediaQueryList: { matches: boolean };
+    mediaQueryListeners: Set<(event: { matches: boolean }) => void>;
+  }[] = [];
+  let breakpoint: Breakpoint = 'lg';
+  if ('initialBreakpoint' in options && options.initialBreakpoint) {
+    breakpoint = options.initialBreakpoint;
+  }
+  const breakpoints: Record<string, Breakpoint> =
+    'queryBreakpointMap' in options &&
+    typeof options.queryBreakpointMap === 'object'
+      ? options.queryBreakpointMap
+      : {
+          '(min-width:1920px)': 'xl',
+          '(min-width:1280px)': 'lg',
+          '(min-width:960px)': 'md',
+          '(min-width:600px)': 'sm',
+          '(min-width:0px)': 'xs',
+        };
   Object.defineProperty(window, 'matchMedia', {
     writable: true,
-    value: jest.fn().mockImplementation(query => ({
-      matches: options.matches ?? false,
-      media: query,
-      onchange: null,
-      addListener: jest.fn(), // deprecated
-      removeListener: jest.fn(), // deprecated
-      addEventListener: jest.fn(),
-      removeEventListener: jest.fn(),
-      dispatchEvent: jest.fn(),
-    })),
+    value: jest.fn().mockImplementation(mediaQueryString => {
+      const mediaQueryListeners = new Set<
+        (event: { matches: boolean }) => void
+      >();
+      const mediaQueryList = {
+        matches:
+          'matches' in options
+            ? options.matches
+            : breakpoints[mediaQueryString] === breakpoint,
+        media: mediaQueryString,
+        onchange: null,
+        addListener: jest.fn(listener => {
+          mediaQueryListeners.add(listener);
+        }),
+        // deprecated
```

Comment:
don't really need these comments tbh, yep they might be but we're reimplementing the API and they're not going anywhere 😅. Think they're on the wrong ones too

Reply:
Removed